### ISSUE TITLE
Getting error while deploying r4 core package downloaded

### ISSUE THREAD

**Issue by arunsundar-m** [brianpos/UploadFIG#8]
I am trying to download the r4 core package from simplifier and run it locally using uploadFIG. Below is the error I am getting

```
HL7 FHIR Implementation Guide Uploader
--------------------------------------
Using local package: C:\Users\XYX\hl7.fhir.r4.core-4.0.1.tgz
MD5 Checksum: 9831228B09EEEE9444F66EF443A4A172

Package dependencies:
Error reading package.json: Value cannot be null. (Parameter 'source')

```
Contents in package.json

```
{
  "name": "hl7.fhir.r4.core",
  "version": "4.0.1",
  "fhir-version-list": [
    "4.0.1"
  ],
  "type": "fhir.core",
  "license": "CC0-1.0",
  "title": "FHIR R4 package : Core",
  "description": "Definitions (API, structures and terminologies) for the R4 version of the FHIR standard",
  "author": "HL7 Inc",
  "url": "http://hl7.org/fhir/R4",
  "tools-version": 3,
  "canonical": "http://hl7.org/fhir",
  "homepage": "http://hl7.org/fhir"
}
```

Faced a similar issue while using microsoft-fhir loader, change the type in package.json and it worked. I would like to check with you before I try this option.

@brianpos Please let me know if there is a way to sort out the issue.

**Comment by brianpos** [comment]
Thanks for reporting the issue, I've spotted a few small bugs that I'll push shortly, but curious which server you're pushing this content to?
Servers built using the Firely SDK will already have all this content "baked" into them and don't need loading.
(Such as the Microsoft FHIR Server)
And I suspect that HAPI is likely the same.

**Comment by brianpos** [comment]
If you update the tool it should no longer have that specific issue that you reported and run through to completion.
_(note that it won't pass through any of the R4 resources that were removed from R4B)_

**Comment by brianpos** [comment]
I've also updated the tool to support R4, R4B and R5. So this now selects which version the package is and then processes using that version.

**Comment by arunsundar-m** [comment]
@brianpos Getting a similar issue while deploying the r4 and r4 examples packages from simplifier
hl7.fhir.r4.core-4.0.1.tgz
hl7.fhir.r4.examples-4.0.1.tgz

```
-----------------------------------

Destination server canonical resource dependency verification:
Argument should be of the form <key>=<value> (Parameter 'criteria')
```
I have updated the tool to the latest version and tried running the command

`UploadFIG -s "C:\Users\XXXX\Downloads\hl7.fhir.r4.core-4.0.1.tgz" -c -d "https://server.fhir.azurehealthcareapis.com" -df json -dh "Authorization:Bearer **** " --verbose >> C:\Users\XXXX\Downloads\hl7.fhir.r4.core-4.0.1_logs.txt`

I am still getting the same issue, can you please help with this?

**Comment by brianpos** [comment]
The server internally contains the r4.core artifacts via it's internal specification.zip.
These are not exposed to the API, but are used for validation.
You also haven't attached the log file it generated to be able to try and assist.

PS. I have also made an update to the `UploadFIG` tool to handle an error when a package doesn't contain any dependencies at all - which the R4 core package doesn't have in it
- Note if you DO load in the default package for R4 they will not have the patches applied to them that are in the server's embedded specification.zip file and some things may not work. There are several search parameters that required technical corrections that Firely have made for them to actually work.

PPS. Ensure that you update your install of UploadFIG too...
``` powershell
PS C:\Users\brianpo> dotnet tool update --global uploadfig
Skipping NuGet package signature verification.
Tool 'uploadfig' was successfully updated from version '2023.12.15.2' to version '2024.1.3.1'.
PS C:\Users\brianpo> uploadfig -pid hl7.fhir.r4.core -t
HL7 FHIR Implementation Guide Uploader
--------------------------------------
Package ID: hl7.fhir.r4.core
Available Versions: 4.0.1
Selecting latest version of package 4.0.1
Package is for FHIR version: R4
Canonical URL: https://packages.simplifier.net/hl7.fhir.r4.core/4.0.1
None.
Direct location: https://packages.simplifier.net/hl7.fhir.r4.core/4.0.1
Downloading hl7.fhir.r4.core|4.0.1 from (FHIR) https://packages.simplifier.net
MD5 Checksum: 9831228B09EEEE9444F66EF443A4A172

Detected FHIR Version 4.0.1 - using 4.0.1

Package dependencies:
    (none)

--------------------------------------

Scanning package content:
    #---> Error validating search parameter http://hl7.org/fhir/SearchParameter/DomainResource-text: DomainResource - _text
      --> error: Search parameter does not contain a fhirpath expression to define its behaviour

    #---> Warning validating search parameter http://hl7.org/fhir/SearchParameter/Observation-component-value-concept: Observation - component-value-concept
      --> warning: Operator 'as' can experience unexpected runtime errors when used with a collection
            Quantity[], CodeableConcept[], string[], boolean[], integer[], Range[], Ratio[], SampledData[], time[], dateTime[], Period[] as string
...
```
